torch/utils/_config_module.py

```diff
@@ -338,11 +363,13 @@ def _make_closure_patcher(self, **changes: Dict[str, Any]) -> Any:
         config = self._config
 
         def change() -> Callable[[], None]:
-            prior = {k: config[k] for k in changes}
-            config.update(changes)
+            prior = {k: config[k].user_override for k in changes}
+            for k, v in changes.items():
+                self._config[k].user_override = v
```

Comment:
This probably slightly regresses critical pass as the previous .update() call was implemented natively in CPython and now we have to do a loop in Python.

Reply:
Let me know if this is a blocker, I can make a "fast path" override dict if that's a concern.

Reply:
Unfortunately we don't seem to microbench this so I can't tell